sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -1078,8 +1077,7 @@ case class GpuFromUTCTimestamp(timestamp: Expression, timezone: Expression)
 
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
     if (rhs.getBase.isValid) {
-      // Just a no-op.
-      lhs.getBase.incRefCount()
+      TimeZoneDB.fromUtcTimestampToTimestamp(lhs.getBase, zoneId)
```

Comment:
A couple of things:

Keep the existing no-op if the timezone is UTC. No need to call into the GPU or CPU timezone db code in that case anyways

Reply:
Updated. Later, we can probably move it within ```TimeZoneDB``` if this touches multiple places.